sql/src/main/java/io/crate/action/sql/BaseResultReceiver.java

```diff
@@ -20,31 +20,38 @@
  * agreement.
  */
 
-package io.crate.concurrent;
+package io.crate.action.sql;
 
+import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.SettableFuture;
+import io.crate.core.collections.Row;
 
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 
-public class ContextCompletionListener implements CompletionListener {
+public class BaseResultReceiver implements ResultReceiver {
```

Comment:
maybe add some javadocs how this class should be used if sub-classes. E.g. which functions are safe to overwrite and which need to call super.
Maybe annotate with `OverridingMethodsMustInvokeSuper` if appropriate.

Should it be abstract or does it make sense to use it as-is without subclassing?